pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -438,41 +442,92 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication {
             revert InvalidUnderlyingTokenAsset();
         }
 
+        // Register asset of wrapper, so it cannot change.
+        _bufferAssets[wrappedToken] = underlyingToken;
+
+        // Take debt for initialization assets.
+        _takeDebt(IERC20(underlyingToken), amountUnderlyingRaw);
+        _takeDebt(wrappedToken, amountWrappedRaw);
+
+        // Update buffer balances.
+        bytes32 bufferBalances = _bufferTokenBalances[wrappedToken];
+        bufferBalances = PackedTokenBalance.toPackedBalance(
+            bufferBalances.getBalanceRaw() + amountUnderlyingRaw,
+            bufferBalances.getBalanceDerived() + amountWrappedRaw
+        );
+
+        _bufferTokenBalances[wrappedToken] = bufferBalances;
+
         // Amount of shares to issue is the total underlying token that the user is depositing.
         issuedShares = wrappedToken.convertToAssets(amountWrappedRaw) + amountUnderlyingRaw;
```

Comment:
Just to mention as auditors got it, `convertToAssets` can be manipulated in a way that returns a big number for any input, including 0. This would make the initializeBuffer() to mint shares for an empty buffer and consider the buffer to be initialized.

Not sure if we want to prevent that, because we'd need to initialize a malicious wrapper in the first place, but maybe we can consider a different "invariant" formula to calculate buffer shares.

Reply:
Maybe we can consider that the initialization is proportional, and then use the proportion of initialization in the addLiquidity. Therefore, we wouldn't need to call an external function to calculate the issuedShares, we'd just use the proportion of underlying and wrapped tokens to calculate the internal rate of the buffer.

Reply:
The main problem that this prevents is a malicious wrapper changing the asset, which causes very bad problems down the line.

If the wrapper is malicious and you're initializing / adding liquidity, you're getting shares for a buffer that doesn't make real sense. We can't prevent that, but all of that should be worthless anyways. As I understand it, the main point is not to mix good wrappers and assets with bad wrappers that can somehow access good assets.

Reply:
Agree with juani, if the wrapper provides a malicious rate, it can only impact it's own balances. The goal here is to ensure isolation of balances.